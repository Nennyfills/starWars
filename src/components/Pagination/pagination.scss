.pagination {
  margin: 1rem 0 4rem 0;
  float: right;
  .button {
    color: #fff;
    float: left;
    padding: 8px 16px;
    text-decoration: none;
    transition: background-color 0.3s;
    border: 1px solid #ddd;
    border-radius: 0.4rem;
    background-color: #884e78;
  }

  .button.active {
    background-color: #4caf50;
    color: white;
    border: 1px solid #4caf50;
  }

  .button:hover:not(.active) {
    background-color: rgb(165, 149, 149);
  }
  .button:disabled,
  .button[disabled] {
    border: 1px solid #999999;
    background-color: #cccccc;
    color: #666666;
  }
  .button:disabled,
  .button:disabled:hover:not(.active) {
    background-color: #cccccc;
  }
}
