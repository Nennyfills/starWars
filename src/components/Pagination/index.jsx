import React from 'react';
import PropTypes from 'prop-types';

import './pagination.scss';
import Button from '../Button';

/**
 * Pagination Component
 *  @param {func} previous
 *  @param {func} next
 *  @param {bool} previousPage
 *  @param {bool} nextPage
 *  @returns pagination element
 */
const Pagination = ({
  previous, next, previousPage, nextPage,
}) => (
  <div className="pagination">
    <Button handleClick={previous} button="button" disabled={!previousPage} name="❮" />
    <Button handleClick={next} button="button" disabled={!nextPage} name="❯" />
  </div>
);

Pagination.propTypes = {
  previous: PropTypes.func,
  next: PropTypes.func,
  previousPage: PropTypes.string,
  nextPage: PropTypes.string,
};
Pagination.defaultProps = {
  previous: () => '',
  next: () => '',
  previousPage: '',
  nextPage: '',
};

export default Pagination;
