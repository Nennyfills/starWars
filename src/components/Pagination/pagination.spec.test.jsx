import React from 'react';

import Pagination from './index';


describe('pagination Component', () => {
  it('should mount the pagination wrapper', () => {
    const wrapper = mount(<Pagination />);

    expect(wrapper.find('.pagination')).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });
});
