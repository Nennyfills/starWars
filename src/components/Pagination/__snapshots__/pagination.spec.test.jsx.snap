// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pagination Component should mount the pagination wrapper 1`] = `
<Pagination>
  <div
    className="pagination"
  >
    <Button
      button="button"
      disabled={true}
      name="❮"
    >
      <input
        className="button"
        disabled={true}
        type="button"
        value="❮"
      />
    </Button>
    <Button
      button="button"
      disabled={true}
      name="❯"
    >
      <input
        className="button"
        disabled={true}
        type="button"
        value="❯"
      />
    </Button>
  </div>
</Pagination>
`;
